packages/shared/src/components/sources/SourceCard.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { ReactElement, ReactNode } from 'react';
+import classNames from 'classnames';
+import { useRouter } from 'next/router';
+import SquadMemberShortList from '../squads/SquadMemberShortList';
+import { Card } from '../cards/Card';
+import { Button } from '../buttons/Button';
+import { SourceType, Squad } from '../../graphql/sources';
+import { Image } from '../image/Image';
+import { SquadJoinButton } from '../squads/SquadJoinButton';
+import { Origin } from '../../lib/analytics';
+import { cloudinary } from '../../lib/image';
+
+type SourceCardActionType = 'link' | 'action';
+
+interface SourceCardAction {
+  type: SourceCardActionType;
+  text: string;
+  href?: string;
+  target?: string;
+  onClick?: () => void;
+}
+
+interface SourceCardProps extends Partial<Squad> {
+  title: string;
+  subtitle?: string;
+  icon?: ReactNode;
+  action?: SourceCardAction;
+}
+
+export enum SourceCardBorderColor {
+  Avocado = 'avocado',
+  Bacon = 'bacon',
+  Cabbage = 'cabbage',
+  Onion = 'onion',
+  Pepper = 'pepper',
+}
+
+const borderColorToClassName: Record<SourceCardBorderColor, string> = {
+  [SourceCardBorderColor.Avocado]:
+    'border-theme-squads-avocado hover:border-theme-squads-avocado',
+  [SourceCardBorderColor.Bacon]:
+    'border-theme-squads-bacon hover:border-theme-squads-bacon',
+  [SourceCardBorderColor.Cabbage]:
+    'border-theme-squads-cabbage hover:border-theme-squads-cabbage',
+  [SourceCardBorderColor.Onion]:
+    'border-theme-squads-onion hover:border-theme-squads-onion',
+  [SourceCardBorderColor.Pepper]:
+    'border-theme-squads-pepper hover:border-theme-squads-pepper',
+};
+
+export const SourceCard = ({
```

Comment:
I meant actually we can simply pass a `source` as one prop so we don't have to manually add all props if that makes sense?
But love to hear @sshanzel thoughts as well.

Reply:
Same sentiments. Would be easier to manage to keep it in a single prop.